### PR TITLE
ZTE-SH-CN-rolling-update-replication-controller.md

### PR DESCRIPTION
**Pull request by Pingan2017**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.8 Features: set Milestone to `1.8` and Base Branch to `release-1.8`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5512)

### CODE REVIEW

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+{:toc}
+
+## 概述
+
+**注**: 创建副本控制器的首选方法是使用[Deployment](/docs/api-reference/{{page.version}}/#deployment-v1beta1-apps)，Deployment使用[ReplicaSet](/docs/api-reference/{{page.version}}/#replicaset-v1beta1-extensions)来进行副本控制。
+更多信息, 查看[使用Deployment运行一个无状态应用](/docs/tasks/run-application/run-stateless-application-deployment/)。
+
+为了在更新服务的同时不中断业务， `kubectl` 支持['滚动更新'](/docs/user-guide/kubectl/v1.6/#rolling-update)，它一次更新一个pod，而不是同时删除整个服务。 有关更多信息，请参阅 [滚动更新设计文档](https://git.k8s.io/community/contributors/design-proposals/simple-rolling-update.md) 和 [滚动更新示例](/docs/tasks/run-application/rolling-update-replication-controller/)。
+
+请注意， `kubectl rolling-update` 仅支持Replication控制器。 但是，如果使用Replication控制器部署应用，请考虑将其切换到[Deployments](/docs/concepts/workloads/controllers/deployment/). Deployment是一种被推荐使用的更高级别的控制器，它可以对应用进行声明性的自动滚动更新。 如果您仍然希望保留您的Replication控制器并使用 `kubectl rolling-update`进行滚动更新， 请继续往下阅读：
```
Comment (tengqm) [inline cn/docs/tasks/run-application/rolling-update-replication-controller.md]: Replication Controller 不能拆开来翻译

Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: Replication Controller不用翻译；直接使用。

Comment (Pingan2017) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: 已修改，谢谢

**Comment by Pingan2017** [comment]
PTAL @tengqm @zhangxiaoyu-zidif

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
+﻿---
+approvers:
+- janetkuo
+title: 对Replication Controller进行滚动升级
```
Comment (tengqm) [inline cn/docs/tasks/run-application/rolling-update-replication-controller.md]: 标题错。请重新理解”Perform Rolling Update Using a Replication Controller“

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+* TOC
+{:toc}
+
+## 概述
+
+**注**: 创建副本控制器的首选方法是使用[Deployment](/docs/api-reference/{{page.version}}/#deployment-v1beta1-apps)，Deployment使用[ReplicaSet](/docs/api-reference/{{page.version}}/#replicaset-v1beta1-extensions)来进行副本控制。
+更多信息, 查看[使用Deployment运行一个无状态应用](/docs/tasks/run-application/run-stateless-application-deployment/)。
+
+为了在更新服务的同时不中断业务， `kubectl` 支持['滚动更新'](/docs/user-guide/kubectl/v1.6/#rolling-update)，它一次更新一个pod，而不是同时删除整个服务。 有关更多信息，请参阅 [滚动更新设计文档](https://git.k8s.io/community/contributors/design-proposals/simple-rolling-update.md) 和 [滚动更新示例](/docs/tasks/run-application/rolling-update-replication-controller/)。
+
```
Comment (tengqm) [inline cn/docs/tasks/run-application/rolling-update-replication-controller.md]: taking down 不是 删除

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    $ kubectl rolling-update NAME \
+        ([NEW_NAME] --image=IMAGE | -f FILE)
+
+## 通过配置文件更新
+
+通过配置文件来进行滚动更新，需要在`kubectl rolling-update`命令后面带上新的配置文件：
+
+    $ kubectl rolling-update NAME -f FILE
+
+这个配置文件必须满足：
```
Comment (tengqm) [inline cn/docs/tasks/run-application/rolling-update-replication-controller.md]: 这个配置文件必须满足以下条件：

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+* 指定不同的`metadata.name`值
+
+* 在`spec.selector`字段中至少有一个共同标签
```
Comment (tengqm) [inline cn/docs/tasks/run-application/rolling-update-replication-controller.md]: 这句完全错了。原文是： Overwrite at least one common label in its spec.selector field.

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+必选字段：
+
+* `NAME`: 需要进行滚动更新的replication controller名称
+
+下面两个字段选其一：
+
+* `-f FILE`: 新的replication controller的配置文件，JSON或者YAML格式均可。配置文件必须指定一个新的顶层`id`值，且至少包含一个现有`spec.selector`中的键值对。
+  详细信息见[通过Replication Controller运行无状态应用](/docs/tutorials/stateless-application/run-stateless-ap-replication-controller/#replication-controller-configuration-file)。
+<br>
+<br>
```
Comment (tengqm) [inline cn/docs/tasks/run-application/rolling-update-replication-controller.md]: 这里的`<br>` 怎么回事？

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md (hunk 2 above)
Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: @tengqm 
yes removing English content is rather difficult for reviewer， let me share my experience:
firstly download English file and Chinese file both
then, use some diff tools like winMerge, svn diff, beyongCompare to compare them both.
lastly, write down comment on the github page.
:)

Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: title: 对Replication Controller进行滚动升级
->
title:基于 Replication Controller执行滚动升级
这篇的关注点是滚动升级。rc是一种表现形式。

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md (hunk 5 above)
Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: Overwrite at least one common label in its spec.selector field
Overwrite 是重写。
翻译中完全没有体现。

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md (hunk 6 above)
Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: @Pingan2017 
please check format by some diff tool before commit.
Thanks

Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: @tengqm 
the original content has two <br>.
So may the auther just want to keep the original format and modify text only.
image

File: cn/docs/tasks/run-application/rolling-update-replication-controller.md (hunk 5 above)
Comment (tengqm) [reply cn/docs/tasks/run-application/rolling-update-replication-controller.md]: 原意是至少要对标签选择算符进行修改，使其与原来的取值不同，这样新的RC和原来的RC才会分别针对不同的Pods集合。现在的翻译把这个语义弄反了。
如果没有英文原文，从译文很难看出问题。

**Comment by Pingan2017** [comment]
PTAL @tengqm

**Pull request merged**